server/routes/index.js

```diff
@@ -1290,4 +1292,73 @@ router.route('/charts/new')
       return res.status(500).send({ message: err.message })
     }
 })
+
+/**
+ * Study Details
+ */
+router.route('/study-details')
+  .get(ensureAuthenticated, async (req, res) => {
+    try {
+      const { display_name, role, icon } = req.session;
+      return res.status(200).send(studyDetailsPage({
+        uid: req.user,
+        name: display_name,
+        role,
+        icon
+      }))
+    } catch (error) {
+      console.error(error.message)
+      return res.status(500).send({ message: err.message})
+    }
+  })
+router.route('/api/v1/study-details/all')
+  .get(ensureAuthenticated, async(req, res) => {
+    try {
+      const studyDetails = await mongoData.collection(collections.studyDetails)
+      .find({ owner: req.user }).toArray()
+      return res.status(200).json({ studyDetails })
+    } catch (error) {
+      return res.status(500).json({ message: error.message })
+    }
+  })
+  router.route('/api/v1/study-details/:detailId')
+  .delete(ensureAuthenticated, async(req, res) =>{
+    const { detailId } = req.params;
+    try {
+      const deleted = await mongoData.collection(collections.studyDetails).deleteOne({
+        _id: ObjectID(detailId)
+      })
+      if (deleted.deletedCount > 0) {
+        return res.status(200).json({ deletedCount: deleted.deletedCount });
+      } else {
+        return res.status(404).json({ message: 'Study Details Not Found' });
+      }
+    } catch (error) {
+      return res.status(500).json({ message: error.message })
+    }
+  })
+router.route('/api/v1/study-details/upload')
+  .post(ensureAuthenticated, async (req, res) => {
+    const { study, owner, targetEnrollment } = req.body
+    try {
+      const inserted = await mongoData
+        .collection(collections.studyDetails)
+        .findOneAndUpdate({ study },
+          { 
+            $set: {
+              study,
+              owner,
+              targetEnrollment,
+              updatedAt: new Date().toISOString()
+            }
+          },
+          {
+            upsert: true
+          }
+        )
+      return res.status(200).json({ inserted })
```

Comment:
I think we should have a consistent response shape. What do you think about something like `body` as the key that has the data we're returning? That way all/most API responses are similar in shape and we can be more consistent in the use of them.

```suggestion
      return res.status(200).json({ body: inserted })
```

Reply:
Going forward it will be data. 👍 
[1731190](https://github.com/AMP-SCZ/dpdash/pull/81/commits/1731190bd6bcc432277da2d2bd9279ea7b81ea5c)